### ISSUE TITLE
Unexpected warning when implementing an interface without @extends

### ISSUE THREAD

**Issue by rictic** [google/closure-compiler#3137]
The following code:

```javascript
/** @interface */
class FooInterface {
  /** @return {string} */
  foo() {};
}


/**
 * @implements {FooInterface}
 */
class Foo {
  /** @return {string} */
  foo() {
    return '';
  }
}
```

Produces the warning:

```
 WARNING - property foo already defined on interface FooInterface; use @override to override it
  foo() {
  ^^^^^^^
```

The documentation at https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#override seems to suggest that this shouldn't be necessary:

> [The `@override` tag] is not required when a class extends a superclass or implements an interface

This is with the internal flag set `RECOMMENDED_FLAGS`. I didn't see a way to repro it on the closure compiler service. Googlers can see cl/220008730 internally, or repro with

      blaze build //experimental/users/rictic/override-implement-interface:bin

**Comment by rictic** [comment]
Things which don't affect this warning:

* class syntax vs function tagged `@constructor`
* `@interface` vs `@record`
* local interface vs one defined in an externs file

**Comment by lauraharker** [comment]
This is WAI, but I agree the documentation is misleading.

> This tag is not required when a class extends a superclass or implements an interface - in those cases the documentation is automatically inherited. This tag is mainly useful to catch typos such as a method named foobar1 in the superclass and mistakenly named foobar2 in the subclass.

The "not required" part really means "not required for type inference to do its job." The wiki should say something like:

> Adding or omitting the `@override` tag does not change what type the compiler infers for a method in a class that extends a superclass or implements an interface. The method's documentation is always automatically inherited from the overridden method in those cases. However, the recommended compiler flagset includes warnings for missing and spurious `@override` annotations. These warnings are mainly useful to catch typos such as a method named foobar1 in the superclass and mistakenly named foobar2 in the subclass

Does that wording make more sense to you? I can update the wiki.

**Comment by rictic** [comment]
Ah, yes that documentation is much better!

**Comment by lauraharker** [comment]
Great! updated https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#override
